charts/opentelemetry-collector/values.yaml

```diff
@@ -304,10 +304,32 @@ topologySpreadConstraints: []
 # Allows for pod scheduler prioritisation
 priorityClassName: ""
 
-extraEnvs: []
-extraEnvsFrom: []
-extraVolumes: []
-extraVolumeMounts: []
+# extraEnvs allows for defining extra environment variables as objects. The key should be the name of the environment variable. See the example below.
+extraEnvs: {}
```

Comment:
I thought it was understood that if we want to do this, users will have adjust any of their own overrides to use objects instead of array.

What do you suggest? Should we have `extraEnvs` and `extraEnvsMap`? Then we don't need to change this to object and can have two arrays instead. But it also seems confusing, how should users know which one to use? My understanding from the original request was we want for users to be able to reuse the same section of YAML to define their own env vars and merge them with ours.

Reply:
I see what you mean know. In that case we can do this with deprecation warning. I'll adjust the PR.